pilot/pkg/xds/eds.go

```diff
@@ -379,28 +379,35 @@ func edsNeedsPush(updates model.XdsUpdates) bool {
 	return false
 }
 
+func allConfigsOfKind(configs map[model.ConfigKey]struct{}, kind string) bool {
+	for k := range configs {
+		if kind != k.Kind.Kind {
+			return false
+		}
+	}
+	return true
+}
+
 func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *model.WatchedResource,
 	req *model.PushRequest) (model.Resources, model.XdsLogDetails, error) {
 	if !edsNeedsPush(req.ConfigsUpdated) {
 		return nil, model.DefaultXdsLogDetails, nil
 	}
 	var edsUpdatedServices map[string]struct{}
-	if !req.Full {
+	if !req.Full ||
+		// all configs updated are ServiceEntry type, then we only need to send eds for these updated services.
```

Comment:
When will we need this check - it seems if it is not a full push we are only sending eds?

Reply:
I want to calculate which eds need to push when se updated.  Envoy does not re-warm clusters that are not changed, so we donot need to push eds for them too?

Reply:
> I want to calculate which eds need to push when se updated

Where are you doing that? edsUpdatedServices has list of all ses?

Reply:
Get all the `updated services` and  

ignore those eds for unchanged clusters: w.ResourceNames - `updated services`

Reply:
sorry. my question was not clear - it did not change with this PR though, we were already doing that, wondering what is the use of this new condition? 

Are you saying this will be used if req is Full and ConfigUpdated has only serviceEntry update even then you want to send eds only?

Not sure if that correct behaviour because if service is changed should n't we send updated cluster?

Reply:
This is working on full push

Reply:
If svc is changed, it will be added to `edsUpdatedServices`

and i meant to gen eds for it